eng/pipelines/docindex.yml

```diff
@@ -33,14 +33,25 @@ jobs:
             - Name: $(DocRepoOwner)/$(DocRepoName)
               WorkingDirectory: $(DocRepoLocation)
 
+      # Set $(DefaultBranch) for use in condition on "Update Docs Onboarding"
+      - template: /eng/common/pipelines/templates/steps/set-default-branch.yml
       # Call update docs ci script to onboard packages
       - task: Powershell@2
         inputs:
           pwsh: true
           filePath: eng/common/scripts/Update-DocsMsPackages.ps1
           arguments: -DocRepoLocation $(DocRepoLocation)
         displayName: Update Docs Onboarding
-
+        condition: >-
+          and(
+            succeeded(),
+            or(
+              eq(variables['Build.Reason'], 'Schedule'),
+              eq(variables['Force.MainUpdate'], 'true'),
+              eq(variables['Build.SourceBranchName'], variables['DefaultBranch'])
```

Comment:
Do we care about the branch of the job we are running on for this? I would expect Schedule with the force override would be enough.

Reply:
I agreed. Our schedule job is currently running on default branch only. Will remove default branch and added when necessary.